src/v/ssx/sugar.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+// TODO: submit upstream if this is useful.
+#include "json/types.h"
+
+#include <seastar/core/semaphore.hh>
+#include <seastar/core/sstring.hh>
+
+#include <fmt/format.h>
+
+#include <cstddef>
+#include <utility>
+
+namespace ssx {
+
+using named_semaphore_units
```

Comment:
I'd be tempted to eliminate the idea of a non-named `semaphore(_units)`, and just call it `ssx::semaphore_units`

Reply:
> I'd be tempted to eliminate the idea of a non-named `semaphore(_units)`, and just call it `ssx::semaphore_units`

This is a spicy :hot_pepper: idea.. I like it. Will resolve this and discuss on other related thread.